src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -449,6 +452,17 @@ private void processIter(DetailAST root, AstState astState) {
         }
     }
 
+    /**
+     * Creates a new {@link TreeSet} with a deterministic order based on the
+     * Check's name before the default ordering.
+     *
+     * @return The new {@link TreeSet}.
+     */
+    private static Set<AbstractCheck> createNewCheckTreeSet() {
```

Comment:
method signature says that it can be any set, not only TreeSet
So either javadoc should be changed or method signature, because they do not match

Reply:
@rnveach may it be `NavigableSet<AbstractCheck>`?

Reply:
Done, made everything `SortedSet` as it makes the most sense.